pandas/core/generic.py

```diff
@@ -6799,6 +6799,9 @@ def interpolate(
 
         obj = self.T if should_transpose else self
 
+        if obj.empty:
```

Comment:
would prefer that this happens in the internal method itself (called on L6861)

Reply:
The error happens before this call (on line 6825)

Reply:
There were a few changes to interpolate in 1.1.0. will run git bisect to ascertain why this is now failing.

Reply:
There was a subtle change in #34752 where

```
if self.ndim == 2 and np.all(self.dtypes == np.dtype(object)):
```
was changed to 
```
if obj.ndim == 2 and np.all(obj.dtypes == np.dtype(object)):
```

does reverting this change restore the old behaviour.

Reply:
the regression is from #33084, but since the error message is about DataFrame columns, I don't think the above should have been changed either. @jbrockmendel

Reply:
> would prefer that this happens in the internal method itself (called on L6861)

As @sanderland notes, the check for all object dtype indeed happens here before calling into the internal method, so the check for empty thus also needs to happen here

However, @sanderland I think the `df.empty` check is not fully correct. Because this was also give True if you have columns but no rows. And for example an empty dataframe with a datetime64[ns] column will give a timedelta64[ns] column as result. That's something we should keep.

So I think we should explicitly check for no columns / no rows (alternatively could also add the check to the offending `if obj.ndim == 2 and np.all(obj.dtypes == np.dtype(object)):`, eg `.. and obj.shape[0]`)

Reply:
@sanderland can you try @jorisvandenbossche suggestion here

Reply:
> However, @sanderland I think the `df.empty` check is not fully correct. Because this was also give True if you have columns but no rows. And for example an empty dataframe with a datetime64[ns] column will give a timedelta64[ns] column as result. That's something we should keep.

I tried this in 1.0.x and it does not, I dont see why interpolate would do this either. Could you suggest a test which fails with my approach but passes in 1.0.x?

Reply:
Hmm, you're fully correct. I might have been mixing up the review of two PRs, as I was also reviewing a regression for `diff()` and maybe was testing that method here as well .. ;) (since `df.diff()` for datetime64 gives timedelta64, also for empty dataframe)

Forget my comment!